test/test_find_if_not.cpp

```diff
@@ -0,0 +1,142 @@
+
+// Copyright (c) 2022 Tristan Brindle (tcbrindle at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <doctest/doctest.h>
+
+#include <array>
+#include <concepts>
+#include <cctype>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#ifdef USE_MODULES
+import flux;
+#else
+#    include <flux/core/default_impls.hpp>
+#    include <flux/algorithm/find.hpp>
+#endif
+
+namespace {
+
+struct S {
+    int i_;
+};
+
+using find_if_not_fn = decltype(flux::find_if_not);
+
+using int_comp = auto(int x) -> bool;
+static_assert(std::invocable<find_if_not_fn, int[10], int_comp>);
+
+using const_int_comp = auto(const int x) -> bool;
+static_assert(std::invocable<find_if_not_fn, int[10], const_int_comp>);
+
+using const_int_ref_comp = auto(const int& x) -> bool;
+static_assert(std::invocable<find_if_not_fn, int[10], const_int_ref_comp>);
+
+// Incompatible predicate type
+using S_comp = auto(S x) -> bool;
+static_assert(not std::invocable<find_if_not_fn, int[10], S_comp>);
+
+// Not a predicate
+using not_a_predicate = auto(int x) -> S;
+static_assert(not std::invocable<find_if_not_fn, int[10], not_a_predicate>);
+
+constexpr bool test_find_if()
```

Comment:
This is testing `find_if_not()` :)

```suggestion
constexpr bool test_find_if_not()
```